### PR TITLE
Sc file size

### PR DESCRIPTION
**Pull request by scoen**
Modify the way getting a file's size works, so that if you use the getFileSize method it will populate the File vertex's fileSize attribute from the blob's reported size if the attribute is null. 

Modified the getSize method (which always tries to get the size from the blob before checking the attribute) so it doesn't call the getFileSize method (as that could call the blob's size method twice). 

I'm not keen on side effects like this change incorporates, but it prevents going to the blob every time you want to know the file's size when calling getFileSize because that can be an expensive/slow activity.

@ato @yetti @josephcurtis @jrixon @sjacob @snezanam @shuangzhou, basically 'at' everyone :-)

### CODE REVIEW

**Comment by yetti** [comment]
What happens if the file gets updated/changed?

File: src/amberdb/model/File.java [hunk 1]
```diff
@@ -251,23 +251,58 @@ private Blob getBlob() throws NoSuchBlobException, IOException {
...
+         */
         @Override
         public long getFileSize() {
             Long fileSize = this.asVertex().getProperty("fileSize");
-            return (fileSize == null)? 0L : fileSize;
+            if (fileSize == null) {
+                try {
+                    fileSize = getBlob().size();
+                    setFileSize(fileSize);
+
```
Comment (shuangzhou) [inline src/amberdb/model/File.java]: @scoen, in amberdb/model/File, we could also rename setBlobId(id) to setBlobIdAndSize(blob), as setBlobId(id) is currently only referenced by 2 methods in the same class: put(...), putWithChecksumValidation(...).  Then in setBlobIdAndSize(), we can call setFileSize(blob.size()).

Comment (scoen) [inline src/amberdb/model/File.java]: good idea - i'll check it out

**Comment by scoen** [comment]
I've addressed Yetrina's issue by making modifications based on Shuang's suggestion and added a couple of tests to check (FileTest.java).

Note: i cleaned up a couple of tests along the way so they extend the AbstractDatabaseIntegrationTest class to make them faster (WorkTest.java and CopyTest.java). These are unrelated to this change except that i saw them and thought i'd do a quick tidy.

@yetti or anyone ?

**Comment by yetti** [comment]
Looks good 👍

**Pull request merged**